### PR TITLE
Add feature example: editable packages

### LINKED ISSUE #42: Add an example for the feature editable packages
Possible references:

https://docs.conan.io/en/latest/developing_packages/editable_packages.html
https://docs.conan.io/en/latest/reference/config_files/editable_layout.html
https://docs.conan.io/en/latest/reference/commands/development/editable.html

Deprecated, it uses `conan link`
https://blog.conan.io/2019/02/04/New-conan-release-1-12.html

https://blog.conan.io/2019/03/12/New-conan-release-1-13.html

Well, I have no power to attribute myself, but I would like to do it.

### PR DESCRIPTION
**Pull request by uilianries**
I've added an example based on workspace feature, as they walk together, I've followed the same lib x app idea

closes #42

### CODE REVIEW

**Comment by uilianries** [comment]
This PR is blocked by https://github.com/conan-io/examples/pull/45

File: features/editable/cmake/.gitignore [hunk 1]
```diff
@@ -0,0 +1,3 @@
+build_debug
```
Comment (jgsogo) [inline features/editable/cmake/.gitignore]: This file works here in a subdirectory? 😮

Comment (uilianries) [reply features/editable/cmake/.gitignore]: actually, I think I've copied by mistake

**Comment by uilianries** [comment]
it failed due multi config example:

```
----Running------

> cmake --build '/Users/travis/build/conan-io/examples/features/multi_config/test_package/build/7336ebca55c566ecc6b383dffaf3ae6eb18d5225' '--' '-j2'

-----------------

Scanning dependencies of target example

[ 50%] Building CXX object CMakeFiles/example.dir/example.cpp.o
[100%] Linking CXX executable bin/example
[100%] Built target example
hello/0.0.1@user/testing (test package): Running test()

ERROR: hello/0.0.1@user/testing (test package): Error in test() method, line 19

	assert ("Hello World {}!".format(str(self.settings.build_type)) in output.getvalue()), output.getvalue()
```

**Comment by uilianries** [comment]
mysteriously, Travis worked after restarting that CI job.

**Comment by uilianries** [comment]
@danimtb @jgsogo all green!

**Comment by uilianries** [comment]
I want to update editable packages on Docs and use that example there. Something similar to workspace

**Comment by jgsogo** [comment]
> I want to update editable packages on Docs and use that example there. Something similar to workspace

Ok, perfect! I was missing that bit of information

**Comment by uilianries** [comment]
The documentation now has an appropriated PR with the new example: https://github.com/conan-io/docs/pull/1567

This PR must be merged first, otherwise, the documentation will be pointing to an invalid example.

File: features/editable/cmake/layout_gcc [hunk 2]
```diff
@@ -0,0 +1,15 @@
+[build_folder]
```
Comment (jgsogo) [inline features/editable/cmake/layout_gcc]: This file should match the documented one: https://github.com/conan-io/docs/pull/1567/files#diff-430fdd2a2f6acd2e29bda5773563cca4R135

File: features/editable/cmake/layout_vs [hunk 3]
```diff
@@ -0,0 +1,12 @@
+[build_folder]
```
Comment (jgsogo) [inline features/editable/cmake/layout_vs]: This file should match the documented one: https://github.com/conan-io/docs/pull/1567/files#diff-430fdd2a2f6acd2e29bda5773563cca4R135

File: features/editable/cmake/say/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+        cmake.configure(source_folder="src")
+        cmake.build()
+
+    def package(self):
+        self.copy("*.h", src="src", dst="include")
+        self.copy("*.lib", dst="lib", keep_path=False)
+        self.copy("*.a", dst="lib", keep_path=False)
+
+    def package_info(self):
+        self.cpp_info.libs = ["say"]
```
[the remark this answers is not in the document]
Comment (uilianries) [reply features/editable/cmake/say/conanfile.py]: I didn't get. What do you mean?

File: features/editable/cmake/layout_gcc (hunk 2 above)
Comment (uilianries) [reply features/editable/cmake/layout_gcc]: only the vs layout is being used there in docs

File: features/editable/cmake/layout_vs (hunk 3 above)
Comment (uilianries) [reply features/editable/cmake/layout_vs]: done

File: features/editable/cmake/say/conanfile.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (uilianries) [reply features/editable/cmake/say/conanfile.py]: The package_info described there in the documentation, are examples how to manipulate the layout, not the recipe that is in the example. They are different things

[the remark this answers is not in the document]
Comment (jgsogo) [reply features/editable/cmake/say/conanfile.py]: If I'm following the docs and they say _"the code can be found in the conan examples repository"_ I really think that we should make them as equal as possible. Adding here the `cpp_info.xxxx = yyyy` lines won't break this example.

File: features/editable/cmake/layout_vs (hunk 3 above)
Comment (jgsogo) [reply features/editable/cmake/layout_vs]: Please, check comment here (https://github.com/conan-io/docs/pull/1567/files#r387704804). It'd be great if this file shows an example of jinja2 replacements (:+1: ) but also of "logic customized for a specific package".

Comment (uilianries) [reply features/editable/cmake/layout_vs]: What do you mean? `{{settings.build_type}}` is Jinja and we are using `Release` as build type, which is indicates a logic behind.

File: features/editable/cmake/say/conanfile.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (uilianries) [reply features/editable/cmake/say/conanfile.py]: The documentation describes 3 ways to use layouts. The example follows the last one, which doesn't affect cpp_info.

File: features/editable/cmake/layout_vs (hunk 3 above)
Comment (uilianries) [reply features/editable/cmake/layout_vs]: I've added a new file, layout_jinja, which has a logic there. The only difference between gcc and visual studio is the build folder.

**Comment by jgsogo** [comment]
Looks like, we have been bitten by this https://github.com/ninja-build/ninja/issues/1599 😭

**Pull request merged**